pkg/controller/erspan.go

```diff
@@ -0,0 +1,428 @@
+	erspanClient, err := erspanclientset.NewForConfig(restconfig)
+	if err != nil {
+		cont.log.Errorf("Failed to intialize erspan client")
+		return
+	}
+	cont.initErspanInformerFromClient(erspanClient)
+	cont.erspanInformer.Run(stopCh)
+}
+
+func podIfInit(cont *AciController, stopCh <-chan struct{}) {
+	cont.log.Debug("Initializing podIf client")
+	restconfig := cont.env.RESTConfig()
+	podIfClient, err := podIfclientset.NewForConfig(restconfig)
+	if err != nil {
+		cont.log.Errorf("Failed to intialize podif client")
+		return
+	}
+	cont.initPodIfInformerFromClient(podIfClient)
+	cont.podIfInformer.Run(stopCh)
+}
+
+func (cont *AciController) initErspanInformerFromClient(
+	erspanClient *erspanclientset.Clientset) {
+	cont.initErspanInformerBase(
+		cache.NewListWatchFromClient(
+			erspanClient.AciV1alpha().RESTClient(), "erspanpolicies",
+			metav1.NamespaceAll, fields.Everything()))
+}
+
+func (cont *AciController) initPodIfInformerFromClient(
+	podIfClient *podIfclientset.Clientset) {
+	cont.initPodIfInformerBase(
+		cache.NewListWatchFromClient(
+			podIfClient.AciV1().RESTClient(), "podifs",
+			metav1.NamespaceAll, fields.Everything()))
+}
+
+func (cont *AciController) initErspanInformerBase(listWatch *cache.ListWatch) {
+	cont.erspanIndexer, cont.erspanInformer = cache.NewIndexerInformer(
+		listWatch, &erspanpolicy.ErspanPolicy{}, 0,
+		cache.ResourceEventHandlerFuncs{
+			AddFunc: func(obj interface{}) {
+				cont.erspanPolicyUpdated(obj)
+			},
+			UpdateFunc: func(_, obj interface{}) {
+				cont.erspanPolicyUpdated(obj)
+			},
+			DeleteFunc: func(obj interface{}) {
+				cont.erspanPolicyDeleted(obj)
+			},
+		},
+		cache.Indexers{cache.NamespaceIndex: cache.MetaNamespaceIndexFunc},
+	)
+	cont.log.Debug("Initializing erspan Policy Informers")
+}
+
+func (cont *AciController) initPodIfInformerBase(listWatch *cache.ListWatch) {
+	cont.podIfIndexer, cont.podIfInformer = cache.NewIndexerInformer(
+		listWatch, &podIfpolicy.PodIF{}, 0,
+		cache.ResourceEventHandlerFuncs{
+			AddFunc: func(obj interface{}) {
+				cont.podIFAdded(obj)
+			},
+			UpdateFunc: func(oldobj interface{}, newobj interface{}) {
+				cont.podIFAdded(newobj)
+			},
+			DeleteFunc: func(obj interface{}) {
+				cont.podIFDeleted(obj)
+			},
+		},
+		cache.Indexers{cache.NamespaceIndex: cache.MetaNamespaceIndexFunc},
+	)
+	cont.log.Debug("Initializing podif Informers")
+}
+
+func getPodifKey(podif *podIfpolicy.PodIF) string {
+	return podif.Status.PodNS + "/" + podif.Status.PodName
+}
+
+func isValidEPG(podif *podIfpolicy.PodIF) bool {
+	pattern := regexp.MustCompile("\\|")
+	idx := pattern.FindAllStringIndex(podif.Status.EPG, -1)
+	if len(idx) != 1 || idx[0][0] == 0 || idx[0][0] == len(podif.Status.EPG)-1 {
+		return false
+	}
+	return true
+}
+
+func (cont *AciController) podIFAdded(obj interface{}) {
+	podif := obj.(*podIfpolicy.PodIF)
+	cont.log.Infof("podif Added: %s", podif.ObjectMeta.Name)
+	cont.indexMutex.Lock()
+	if isValidEPG(podif) {
```

Comment:
can we fix this issue if there is any EPG issue before populating the podif's

Reply:
That makes sense, will remove the check here.